packages/@ama-sdk/core/src/plugins/abort/abort.fetch.ts

```diff
@@ -0,0 +1,54 @@
+import type { FetchCall, FetchPlugin, FetchPluginContext, RequestOptions } from '../core';
+
+interface AbortCallbackParameters {
+  /** URL targeted */
+  url: string;
+
+  /** Fetch call options */
+  options: RequestInit | RequestOptions;
+
+  // TODO Now supported for all the modern browsers - should become mandatory in @ama-sdk/core@10.0
+  /** Abort controller to abort fetch call */
+  controller?: AbortController;
+}
+
+const isPromise = (result: boolean | void | Promise<void> | Promise<boolean>): result is (Promise<void> | Promise<boolean>) => {
+  if (typeof result !== 'object') {
```

Comment:
Is this check enough and future proof? Could we check something like an exisiting catch or then property?